app/cdap/components/ConnectionList/index.tsx

```diff
@@ -46,228 +90,294 @@ const SelectDatasetWrapper = styled(Box)({
   },
 });
 
-export default function ConnectionList() {
-  const { connectorType } = useParams() as IRecords;
+const TabsContainerWithHeader = styled(FlexContainer)`
+  flex-direction: column;
+  border-right: 1px solid ${grey[300]};
+`;
 
-  const refs = useRef([]);
-  const classes = useStyles();
+const TabHeaderContainer = styled(Box)`
+  background-color: ${blue[50]};
+  & .MuiTypography-root{
+    font-size: 16px;
+    color: #000000;
+    },    
+`;
+
+export default function() {
+  const { connectorType } = useParams() as IRecords;
   const loc = useLocation();
   const queryParams = new URLSearchParams(loc.search);
   const pathFromUrl = queryParams.get('path') || '/';
-  const [loading, setLoading] = useState(true);
+  const refs = useRef([]);
+  const headersRefs = useRef([]);
+  const [loading, setLoading] = useState<boolean>(true);
   const [isErrorOnNoWorkspace, setIsErrorOnNoWorkSpace] = useState<boolean>(false);
-
-  const toggleLoader = (value: boolean, isError?: boolean) => {
-    setLoading(value);
-  };
-  let connectionId;
-  const [dataForTabs, setDataForTabs] = useState([
+  const [tabsLength, setTabsLength] = useState<number>(0);
+  const [tabsData, setTabsData] = useState<IFilteredData[]>([
     {
       data: [],
       showTabs: true,
       selectedTab: null,
-      isSearching: false,
+      toggleSearch: false,
     },
   ]);
+  const [filteredData, setFilteredData] = useState<IFilteredData>(cloneDeep(tabsData));
+  const toggleLoader = (value: boolean, isError?: boolean) => {
+    setLoading(value);
+  };
+  const [fetchedConnectorsData, setFetchedConnectorsData] = useState([]);
+  let connectionId;
+
+  useEffect(() => {
+    getWidgetData();
+  }, []);
+
+  /**
+   * Here we are fetching connectors with icons from store
+   */
+  DataPrepStore.subscribe(() => {
+    const newState = DataPrepStore.getState();
+    setFetchedConnectorsData(newState.dataprep.connectorsWithIcons);
+  });
 
   const getConnectionsTabData = async () => {
-    // Fetching the all available connectors list
-    let connectorTypes = await fetchConnectors();
+    let connectorTypes: ITabData[] = [];
```

Comment:
Why not `let connectorTypes = fetchedConnectorsData`?

Reply:
we modified the code to reflect the above suggestion.